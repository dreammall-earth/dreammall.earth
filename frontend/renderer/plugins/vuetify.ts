// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { ThemeDefinition, createVuetify } from 'vuetify'
import { aliases as mdiAliases } from 'vuetify/iconsets/mdi'
import { VNumberInput } from 'vuetify/labs/VNumberInput'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

import { aliases as customAliases } from '#assets/icons'
import dark from '#assets/scss/dark.module.scss'
import light from '#assets/scss/light.module.scss'

function makeThemeFromCssModule(theme: CSSModuleClasses, isDark: boolean): ThemeDefinition {
  return {
    dark: isDark,
    colors: {
      background: theme.backgroundColor,
      surface: theme.surfaceColor,
      primary: theme.primaryColor,
      secondary: theme.secondaryColor,
      success: theme.successColor,
      info: theme.infoColor,
      warning: theme.warningColor,
      error: theme.errorColor,
      font: theme.fontColor,
      'dropdown-background': theme.dropdownBackgroundColor,
      'cockpit-highlight': theme.cockpitHighlightColor,
      'dm-panel-submit-color': theme.dmPanelSubmitColor,
      'dm-panel-submit-background-color': theme.dmPanelSubmitBackgroundColor,
      'dm-panel-text-input-color': theme.dmPanelTextInputColor,
      'dm-panel-call-action-button-color': theme.dmPanelCallActionButtonColor,
      'dm-panel-call-action-button-background-color': theme.dmPanelCallActionButtonBackgroundColor,
      'dm-panel-call-action-button-indicator-color': theme.dmPanelCallActionButtonIndicatorColor,
      'dm-panel-call-action-button-indicator-background-color':
        theme.dmPanelCallActionButtonIndicatorBackgroundColor,
      'dm-panel-reminder-text-color': theme.dmPanelReminderTextColor,
      'dm-panel-reminder-link-color': theme.dmPanelReminderLinkColor,
      'dm-switch-inactive-color': theme.dmSwitchInactiveColor,
      'dm-switch-active-color': theme.dmSwitchActiveColor,
    },
    variables: {
      'border-color': theme.borderColor,
      'border-opacity': theme.borderOpacity,
      // Vuetify colors don't support opacity, so we use variables instead.
      // Beware that HEX values won't work.
      'icon-background': theme.iconBackground,
      'bottom-menu-background': theme.bottomMenuBackground,
      'sidebar-background': theme.sidebarBackground,
      'cockpit-input-background': theme.cockpitInputBackground,
      'cockpit-element-background': theme.cockpitElementBackground,
      'cockpit-chip-background': theme.cockpitChipBackground,
      'cockpit-chip-background-2': theme.cockpitChipBackground2,
      'panel-background': theme.panelBackground,
      'panel-element-background': theme.panelElementBackground,
      'dm-panel-text-input-background-color': theme.dmPanelTextInputBackgroundColor,
      'dm-panel-background-color': theme.dmPanelBackgroundColor,
      'dm-panel-border-color': theme.dmPanelBorderColor,
      'dm-panel-reminder-text-background-color': theme.dmPanelReminderTextBackgroundColor,
      'dm-panel-reminder-link-background-color': theme.dmPanelReminderLinkBackgroundColor,
    },
  }
}

const themes = {
  light: makeThemeFromCssModule(light, false),
  dark: makeThemeFromCssModule(dark, true),
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    components: {
      VNumberInput,
    },
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
    icons: {
      aliases: { ...mdiAliases, ...customAliases },
    },
    theme: {
      defaultTheme: 'light',
      themes,
    },
  })
