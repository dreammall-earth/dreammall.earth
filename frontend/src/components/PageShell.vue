<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
