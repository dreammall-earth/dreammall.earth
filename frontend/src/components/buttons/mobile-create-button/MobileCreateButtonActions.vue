<template>
  <div
    class="button-list-mobile d-md-none"
    :class="[isVisible ? 'button-list-mobile--active' : '']"
  >
    <v-img class="w-100 menu-divider" :src="Divider" />
    <v-img
      class="w-100 menu-triangle"
      :class="[isVisible ? 'menu-triangle--turned' : '']"
      :src="Triangle"
    />
    <MainButton
      class="assistant-button"
      variant="border-gradient"
      label="Assistant"
      size="auto"
      icon="ear-hearing"
      >{{ $t('buttons.toAssistant') }}
    </MainButton>
    <MainButton
      class="new-table-button"
      variant="border-yellow"
      label="New Table"
      size="auto"
      icon="plus"
      @click="enterRoom"
      >{{ $t('buttons.newTable') }}
    </MainButton>
    <MainButton
      class="new-project-button"
      variant="border-blue"
      label="New Project"
      size="auto"
      icon="plus"
      >{{ $t('buttons.newProject') }}
    </MainButton>
  </div>
</template>

<script lang="ts" setup>
import { useMutation } from '@vue/apollo-composable'
import { navigate } from 'vike/client/router'

import Divider from '#assets/img/divider.svg'
import Triangle from '#assets/img/triangle.svg'
import MainButton from '#components/buttons/MainButton.vue'
import { JoinMyRoomMutationResult, joinMyRoomMutation } from '#mutations/joinMyRoomMutation'
import GlobalErrorHandler from '#plugins/globalErrorHandler'
import { useActiveRoomStore } from '#stores/activeRoomStore'

defineProps<{
  isVisible: boolean
}>()

const activeRoomStore = useActiveRoomStore()

const { mutate: joinMyRoomMutationResult } = useMutation<JoinMyRoomMutationResult>(
  joinMyRoomMutation,
  {
    fetchPolicy: 'no-cache',
  },
)

const enterRoom = async () => {
  try {
    const result = await joinMyRoomMutationResult()

    if (result?.data?.joinMyRoom) {
      activeRoomStore.setActiveRoom(result.data.joinMyRoom)
      navigate('/room/')
    } else {
      GlobalErrorHandler.error('No room found')
    }
  } catch (error) {
    GlobalErrorHandler.error('Error opening room', error)
  }
}
</script>

<style scoped lang="scss">
.button-list-mobile {
  --height: 220px;
  --width: 180px;

  position: fixed;
  bottom: calc(var(--height) * -1);
  left: calc(50% - var(--width) / 2);
  z-index: 1;
  display: flex;
  flex-direction: column;
  gap: 15px;
  align-items: center;
  justify-content: start;
  width: var(--width);
  height: var(--height);
  padding-top: 30px;
  background: var(--v-bottom-menu-background) !important;
  backdrop-filter: blur(20px);
  border-radius: 30px 30px 0 0;
  transition: bottom 0.75s;

  &--active {
    bottom: 60px;
  }

  .menu-triangle {
    position: absolute;
    top: -95px;
    max-width: 10px;
    transition:
      0.75s transform,
      0.75s 0.25s top;
    transform-origin: center;

    &--turned {
      top: 10px;
      transition:
        0.75s rotate,
        0.5s top;
      transform: rotate(180deg);
    }
  }

  .menu-divider {
    position: absolute;
    top: 0;
    max-width: 60px;
  }

  .assistant-button {
    margin: 0 40px;
    transition-delay: 0.2s;

    :deep(i) {
      margin-right: 16px;
    }
  }

  .new-project-button {
    margin: 0 20px;
    transition-delay: 0s;
  }

  .new-table-button {
    margin: 0 20px;
    transition-delay: 0.1s;
  }
}
</style>
