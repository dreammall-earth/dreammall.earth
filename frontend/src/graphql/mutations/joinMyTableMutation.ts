import { gql } from 'graphql-tag'

export const joinMyTableMutation = gql`
  mutation {
    joinMyTable
  }
`

export type JoinMyTableMutationResult = {
  joinMyTable: string
}
