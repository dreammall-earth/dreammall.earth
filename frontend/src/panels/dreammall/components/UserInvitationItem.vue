<template>
  <v-list-item class="user-invitation-item my-2 pa-2 pr-3 rounded-pill" elevation="0" rounded>
    <template #prepend>
      <v-avatar size="40">
        <v-img v-if="user.avatar" :src="user.avatar" :alt="user.name" cover></v-img>
        <svg v-else width="40" height="40" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
          <circle cx="20" cy="20" r="20" fill="#E0E0E0" />
          <circle cx="20" cy="15" r="7" fill="#BDBDBD" />
          <path
            d="M6 34.5C6 27.5964 11.5964 22 18.5 22H21.5C28.4036 22 34 27.5964 34 34.5V40H6V34.5Z"
            fill="#BDBDBD"
          />
        </svg>
      </v-avatar>
    </template>

    <v-list-item-title class="text-white font-weight-medium">
      {{ user.name }}
    </v-list-item-title>

    <template #append>
      <v-btn
        elevation="0"
        :color="user.invited ? 'success' : 'secondary'"
        class="invitation-button text-caption py-2 px-3 rounded-pill"
        size="small"
        @click="toggleInvitation"
      >
        {{
          user.invited
            ? $t('dream-mall-panel.setup.invitation-sent')
            : $t('dream-mall-panel.setup.invite')
        }}
      </v-btn>
    </template>
  </v-list-item>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits } from 'vue'

import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const props = defineProps<{
  user: UserInvitation
}>()

const emit = defineEmits<{
  (e: 'update:invited', value: boolean): void
}>()

const toggleInvitation = () => {
  emit('update:invited', !props.user.invited)
}
</script>

<style lang="scss" scoped>
.user-invitation-item {
  --background-user-item: #5d6670; // todo: save globally
  --background-invitation-button: #8b949b; // todo: save globally

  background-color: var(--background-user-item) !important;

  .invitation-button {
    &.bg-secondary {
      background-color: var(--background-invitation-button) !important;
    }
  }
}
</style>
