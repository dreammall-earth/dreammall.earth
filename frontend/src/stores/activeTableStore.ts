import { acceptHMRUpdate, defineStore } from 'pinia'
import { computed, ref } from 'vue'

export const useActiveTableStore = defineStore(
  'activeTable',
  () => {
    const activeTable = ref<string | null>(null)

    const getActiveTable = computed(() => activeTable.value)

    const setActiveTable = (table: string | null) => {
      activeTable.value = table
    }

    return {
      activeTable,
      setActiveTable,
      getActiveTable,
    }
  },
  {
    persist: true,
  },
)

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useActiveTableStore, import.meta.hot))
}
