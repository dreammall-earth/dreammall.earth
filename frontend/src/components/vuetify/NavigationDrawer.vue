<template>
  <v-navigation-drawer v-model="drawer" :location="computedLocation" temporary width="400">
    <slot />
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { computed, ref, watch } from 'vue'
import { useDisplay } from 'vuetify'

export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)

const emits = defineEmits(['update:modelValue'])
const display = useDisplay()
const drawer = ref(props.modelValue)

const computedLocation = computed(() => {
  return display.mobile.value ? 'bottom' : props.location
})

watch(
  () => props.modelValue,
  (newVal) => {
    drawer.value = newVal
  },
)

watch(
  () => drawer.value,
  (newVal) => {
    if (newVal !== props.modelValue) {
      emits('update:modelValue', newVal)
    }
  },
)
</script>
