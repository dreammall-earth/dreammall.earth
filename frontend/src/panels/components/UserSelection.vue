<template>
  <v-text-field
    v-model="userSearch"
    rounded
    clearable
    class="elevation-0 w-100"
    content-class="elevation-0"
    label="Suche"
    variant="solo-filled"
    append-inner-icon="mdi-magnify"
  />

  <div class="user-list-container w-100">
    <div v-if="error" class="text-center text-error">
      {{ $t('dream-mall-panel.setup.search-error') }}
    </div>

    <v-list
      v-if="displayedUsers.length"
      class="bg-transparent w-100 user-list"
      :class="{ loading: isLoading }"
    >
      <UserInvitationItem
        v-for="user in displayedUsers"
        :key="user.id"
        :user="user"
        @update:invited="updateInvitationStatus(user.id, $event)"
      />
    </v-list>

    <div v-if="!displayedUsers.length" class="text-center mt-4 text-grey-darken-2">
      {{ $t('dream-mall-panel.setup.no-result') }}
    </div>
  </div>
</template>

<script lang="ts" setup>
import { ref, computed, watch } from 'vue'

import UserInvitationItem from '#src/panels/components/UserInvitationItem.vue'
import useSearchUsers from '#src/panels/composables/useSearchUsers'
import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const users = defineModel<number[]>({ required: true })

const { searchUsers, searchResults, isLoading, error } = useSearchUsers()
const userSearch = ref('')

watch(userSearch, (newSearch) => {
  searchUsers(newSearch)
})

const displayedUsers = computed<UserInvitation[]>(() => {
  return searchResults.value.map((user) => ({
    id: user.id,
    name: user.name || user.username,
    avatar: null, // not supporting yet
    invited: users.value.includes(user.id), // todo: user already included in myTable?
  }))
})

const updateInvitationStatus = (userId: number, invited: boolean) => {
  let currentUsers: number[] = [...users.value]
  if (invited && !currentUsers.includes(userId)) {
    currentUsers.push(userId)
  } else if (!invited && currentUsers.includes(userId)) {
    currentUsers = currentUsers.filter((id) => id !== userId)
  }
  users.value = currentUsers
}
</script>

<style lang="scss" scoped>
.flat-text-field .v-field--variant-solo-filled {
  box-shadow: none !important;
}

.user-list-container {
  height: 200px;
  overflow-y: auto;

  .user-list {
    height: 100%;

    &.loading {
      opacity: 0.5;
      transition: opacity 0.5s ease;
    }
  }
}
</style>
