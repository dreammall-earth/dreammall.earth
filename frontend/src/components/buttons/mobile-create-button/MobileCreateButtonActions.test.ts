import { ApolloError } from '@apollo/client/errors'
import { flushPromises, mount } from '@vue/test-utils'
import { navigate } from 'vike/client/router'
import { describe, it, expect, beforeEach, vi, afterEach } from 'vitest'
import { h } from 'vue'
import { VApp } from 'vuetify/components'

import { joinMyTableMutation } from '#mutations/joinMyTableMutation'
import { useActiveTableStore } from '#stores/activeTableStore'
import { mockClient } from '#tests/mock.apolloClient'
import { errorHandlerSpy } from '#tests/plugin.globalErrorHandler'

import MobileCreateButtonActions from './MobileCreateButtonActions.vue'

vi.mock('vike/client/router')
vi.mocked(navigate).mockResolvedValue()

const joinMyTableMutationMock = vi.fn()

mockClient.setRequestHandler(joinMyTableMutation, joinMyTableMutationMock)

const activeTableStore = useActiveTableStore()

describe('MobileCreateButtonActions', () => {
  const Wrapper = (props = { isVisible: true }) => {
    return mount(VApp, {
      slots: {
        default: h(MobileCreateButtonActions, props),
      },
    })
  }
  let wrapper: ReturnType<typeof Wrapper>

  afterEach(() => {
    wrapper.unmount()
  })

  it('renders open', () => {
    wrapper = Wrapper()
    expect(wrapper.findComponent(MobileCreateButtonActions).element).toMatchSnapshot()
  })

  it('renders closed', () => {
    wrapper = Wrapper({ isVisible: false })
    expect(wrapper.findComponent(MobileCreateButtonActions).element).toMatchSnapshot()
  })

  describe('new table button', () => {
    beforeEach(() => {
      wrapper = Wrapper()
    })

    describe('enter table', () => {
      describe('apollo with success', () => {
        beforeEach(async () => {
          vi.clearAllMocks()
          joinMyTableMutationMock.mockResolvedValue({
            data: {
              joinMyTable: 'http://link-to-my.table',
            },
          })
          await wrapper.find('button.new-table-button').trigger('click')
        })

        it('calls the api', () => {
          // eslint-disable-next-line vitest/prefer-called-with
          expect(joinMyTableMutationMock).toHaveBeenCalled()
        })

        it('updates the store', () => {
          expect(activeTableStore.activeTable).toBe('http://link-to-my.table')
        })

        it('navigates to table page', async () => {
          await flushPromises()
          expect(navigate).toHaveBeenCalledWith('/table/')
        })
      })

      describe('apollo with no data', () => {
        beforeEach(async () => {
          activeTableStore.setActiveTable(null)
          vi.clearAllMocks()
          joinMyTableMutationMock.mockResolvedValue({
            data: null,
          })
          await wrapper.find('.button-list-mobile button.new-table-button').trigger('click')
        })

        it('calls the api', () => {
          // eslint-disable-next-line vitest/prefer-called-with
          expect(joinMyTableMutationMock).toHaveBeenCalled()
        })

        it('does not update the store', () => {
          expect(activeTableStore.activeTable).toBeNull()
        })

        it('toasts no table found error', async () => {
          await flushPromises()
          expect(errorHandlerSpy).toHaveBeenCalledWith('No table found')
        })
      })

      describe('apollo with error', () => {
        beforeEach(async () => {
          activeTableStore.setActiveTable(null)
          vi.clearAllMocks()
          joinMyTableMutationMock.mockRejectedValue({
            message: 'OUCH',
          })
          await wrapper.find('.button-list-mobile button.new-table-button').trigger('click')
        })

        it('calls the api', () => {
          // eslint-disable-next-line vitest/prefer-called-with
          expect(joinMyTableMutationMock).toHaveBeenCalled()
        })

        it('does not update the store', () => {
          expect(activeTableStore.activeTable).toBeNull()
        })

        it('toasts no table found error', () => {
          expect(errorHandlerSpy).toHaveBeenCalledWith(
            'Error opening table',
            new ApolloError({ errorMessage: 'OUCH' }),
          )
        })
      })
    })
  })
})
