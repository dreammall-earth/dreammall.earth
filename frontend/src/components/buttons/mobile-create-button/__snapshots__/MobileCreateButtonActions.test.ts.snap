// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileCreateButtonActions > renders closed 1`] = `
<div
  class="button-list-mobile d-md-none"
  data-v-1479d999=""
>
  <div
    class="v-responsive v-img v-img--booting w-100 menu-divider"
    data-v-1479d999=""
  >
    <div
      class="v-responsive__sizer"
    />
    
    
    <transition-stub
      appear="true"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <img
        class="v-img__img v-img__img--contain"
        src="/src/assets/img/divider.svg"
        style="display: none;"
      />
    </transition-stub>
    <transition-stub
      appear="false"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <!---->
    </transition-stub>
    <!---->
    <!---->
    <!---->
    
    
    <!---->
  </div>
  <div
    class="v-responsive v-img v-img--booting w-100 menu-triangle"
    data-v-1479d999=""
  >
    <div
      class="v-responsive__sizer"
    />
    
    
    <transition-stub
      appear="true"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <img
        class="v-img__img v-img__img--contain"
        src="/src/assets/img/triangle.svg"
        style="display: none;"
      />
    </transition-stub>
    <transition-stub
      appear="false"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <!---->
    </transition-stub>
    <!---->
    <!---->
    <!---->
    
    
    <!---->
  </div>
  <button
    class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--border-gradient main-button-auto assistant-button"
    data-v-1479d999=""
    data-v-2b63c1fe=""
    style="background-color: #f09630; color: #000; caret-color: #000; height: auto; width: auto;"
    type="button"
  >
    
    <span
      class="v-btn__overlay"
    />
    <span
      class="v-btn__underlay"
    />
    
    <!---->
    <span
      class="v-btn__content"
      data-no-activator=""
    >
      
      
      <!--v-if-->
      <i
        aria-hidden="true"
        class="mdi-ear-hearing mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--start icon pb-1"
        data-v-2b63c1fe=""
      />
      <span
        class="main-button-content"
        data-v-2b63c1fe=""
      >
        Assistant 
        <!--v-if-->
      </span>
      
      
    </span>
    <!---->
    <!---->
  </button>
  <button
    class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--border-yellow main-button-auto new-table-button"
    data-v-1479d999=""
    data-v-2b63c1fe=""
    style="background-color: #f09630; color: #000; caret-color: #000; height: auto; width: auto;"
    type="button"
  >
    
    <span
      class="v-btn__overlay"
    />
    <span
      class="v-btn__underlay"
    />
    
    <!---->
    <span
      class="v-btn__content"
      data-no-activator=""
    >
      
      
      <!--v-if-->
      <i
        aria-hidden="true"
        class="mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--start icon pb-1"
        data-v-2b63c1fe=""
      />
      <span
        class="main-button-content"
        data-v-2b63c1fe=""
      >
        New Table 
        <!--v-if-->
      </span>
      
      
    </span>
    <!---->
    <!---->
  </button>
  <button
    class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--border-blue main-button-auto new-project-button"
    data-v-1479d999=""
    data-v-2b63c1fe=""
    style="background-color: #f09630; color: #000; caret-color: #000; height: auto; width: auto;"
    type="button"
  >
    
    <span
      class="v-btn__overlay"
    />
    <span
      class="v-btn__underlay"
    />
    
    <!---->
    <span
      class="v-btn__content"
      data-no-activator=""
    >
      
      
      <!--v-if-->
      <i
        aria-hidden="true"
        class="mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--start icon pb-1"
        data-v-2b63c1fe=""
      />
      <span
        class="main-button-content"
        data-v-2b63c1fe=""
      >
        New Project 
        <!--v-if-->
      </span>
      
      
    </span>
    <!---->
    <!---->
  </button>
</div>
`;

exports[`MobileCreateButtonActions > renders open 1`] = `
<div
  class="button-list-mobile d-md-none button-list-mobile--active"
  data-v-1479d999=""
>
  <div
    class="v-responsive v-img v-img--booting w-100 menu-divider"
    data-v-1479d999=""
  >
    <div
      class="v-responsive__sizer"
    />
    
    
    <transition-stub
      appear="true"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <img
        class="v-img__img v-img__img--contain"
        src="/src/assets/img/divider.svg"
        style="display: none;"
      />
    </transition-stub>
    <transition-stub
      appear="false"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <!---->
    </transition-stub>
    <!---->
    <!---->
    <!---->
    
    
    <!---->
  </div>
  <div
    class="v-responsive v-img v-img--booting w-100 menu-triangle menu-triangle--turned"
    data-v-1479d999=""
  >
    <div
      class="v-responsive__sizer"
    />
    
    
    <transition-stub
      appear="true"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <img
        class="v-img__img v-img__img--contain"
        src="/src/assets/img/triangle.svg"
        style="display: none;"
      />
    </transition-stub>
    <transition-stub
      appear="false"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <!---->
    </transition-stub>
    <!---->
    <!---->
    <!---->
    
    
    <!---->
  </div>
  <button
    class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--border-gradient main-button-auto assistant-button"
    data-v-1479d999=""
    data-v-2b63c1fe=""
    style="background-color: #f09630; color: #000; caret-color: #000; height: auto; width: auto;"
    type="button"
  >
    
    <span
      class="v-btn__overlay"
    />
    <span
      class="v-btn__underlay"
    />
    
    <!---->
    <span
      class="v-btn__content"
      data-no-activator=""
    >
      
      
      <!--v-if-->
      <i
        aria-hidden="true"
        class="mdi-ear-hearing mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--start icon pb-1"
        data-v-2b63c1fe=""
      />
      <span
        class="main-button-content"
        data-v-2b63c1fe=""
      >
        Assistant 
        <!--v-if-->
      </span>
      
      
    </span>
    <!---->
    <!---->
  </button>
  <button
    class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--border-yellow main-button-auto new-table-button"
    data-v-1479d999=""
    data-v-2b63c1fe=""
    style="background-color: #f09630; color: #000; caret-color: #000; height: auto; width: auto;"
    type="button"
  >
    
    <span
      class="v-btn__overlay"
    />
    <span
      class="v-btn__underlay"
    />
    
    <!---->
    <span
      class="v-btn__content"
      data-no-activator=""
    >
      
      
      <!--v-if-->
      <i
        aria-hidden="true"
        class="mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--start icon pb-1"
        data-v-2b63c1fe=""
      />
      <span
        class="main-button-content"
        data-v-2b63c1fe=""
      >
        New Table 
        <!--v-if-->
      </span>
      
      
    </span>
    <!---->
    <!---->
  </button>
  <button
    class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--border-blue main-button-auto new-project-button"
    data-v-1479d999=""
    data-v-2b63c1fe=""
    style="background-color: #f09630; color: #000; caret-color: #000; height: auto; width: auto;"
    type="button"
  >
    
    <span
      class="v-btn__overlay"
    />
    <span
      class="v-btn__underlay"
    />
    
    <!---->
    <span
      class="v-btn__content"
      data-no-activator=""
    >
      
      
      <!--v-if-->
      <i
        aria-hidden="true"
        class="mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--start icon pb-1"
        data-v-2b63c1fe=""
      />
      <span
        class="main-button-content"
        data-v-2b63c1fe=""
      >
        New Project 
        <!--v-if-->
      </span>
      
      
    </span>
    <!---->
    <!---->
  </button>
</div>
`;
